### PR TITLE
Upgrade ember to version 3.3.0

### PR DESCRIPTION
**Pull request by Haz901**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,10 +24,9 @@ branches:
 
 env:
   - EMBER_TRY_SCENARIO=ember-default
-  - EMBER_TRY_SCENARIO=ember-lts-2.4
-  - EMBER_TRY_SCENARIO=ember-lts-2.8
```
Comment (michaelw8) [inline .travis.yml]: Why did you remove this?

File: .eslintrc.js [hunk 2]
```diff
@@ -28,5 +28,33 @@ module.exports = {
...
+        'blueprints/*/index.js',
+        'config/**/*.js',
+        'tests/dummy/config/**/*.js'
+      ],
+      excludedFiles: [
+        'addon/**',
+        'addon-test-support/**',
+        'app/**',
+        'tests/dummy/app/**'
+      ],
```
Comment (michaelw8) [inline .eslintrc.js]: What does this mean? Exclude files from eslint?

File: tests/unit/components/data-grid-test.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ test('with content: has a ArrayDataManager with content', function(assert) {
     dataGrid = this.subject({
       content: [1, 2, 3, 4]
     });
+    dataGrid.didReceiveAttrs();
```
Comment (michaelw8) [inline tests/unit/components/data-grid-test.js]: Why do you need this?

File: testem.js [hunk 4]
```diff
@@ -10,13 +10,14 @@ module.exports = {
...
   browser_args: {
     Chrome: {
-      mode: 'ci',
-      args: [
+      ci: [
         // --no-sandbox is needed when running Chrome inside a container
-        process.env.TRAVIS ? '--no-sandbox' : null,
-
-        '--disable-gpu',
+        process.env.CI ? '--no-sandbox' : null,
```
Comment (michaelw8) [inline testem.js]: Does this change anything?

File: .eslintrc.js (hunk 2 above)
Comment (Haz901) [reply .eslintrc.js]: I guess so, these are changes that were introduced by the ember update

File: .travis.yml (hunk 1 above)
Comment (Haz901) [reply .travis.yml]: Not manually removed, ember init changed this

File: tests/unit/components/data-grid-test.js (hunk 3 above)
Comment (Haz901) [reply tests/unit/components/data-grid-test.js]: Because this is no longer automatically done by the constructor in unit tests

File: testem.js (hunk 4 above)
Comment (Haz901) [reply testem.js]: Don't know, ember init did this too

File: .travis.yml (hunk 1 above)
Comment (Haz901) [reply .travis.yml]: Ember-source 2.4 is no longer available

File: .eslintrc.js [hunk 5]
```diff
@@ -28,5 +28,33 @@ module.exports = {
...
+      parserOptions: {
+        sourceType: 'script',
+        ecmaVersion: 2015
```
Comment (cbroeren) [inline .eslintrc.js]: I'm not sure about this

File: .travis.yml [hunk 6]
```diff
@@ -3,34 +3,36 @@ language: node_js
 node_js:
   - "6.9.1"
 
+sudo: false
```
Comment (cbroeren) [inline .travis.yml]: Hope this will work, because there was a reason we changed it to true

File: .travis.yml [hunk 7]
```diff
@@ -3,34 +3,36 @@ language: node_js
...
+    # as well as latest stable release (bonus points to beta/canary)
+    - EMBER_TRY_SCENARIO=ember-lts-2.4
+    - EMBER_TRY_SCENARIO=ember-lts-2.8
+    - EMBER_TRY_SCENARIO=ember-lts-2.12
+    - EMBER_TRY_SCENARIO=ember-lts-2.16
+    - EMBER_TRY_SCENARIO=ember-lts-2.18
+    - EMBER_TRY_SCENARIO=ember-release
+    - EMBER_TRY_SCENARIO=ember-beta
+    - EMBER_TRY_SCENARIO=ember-canary
+    - EMBER_TRY_SCENARIO=ember-default
```
Comment (cbroeren) [inline .travis.yml]: Move the default back to the top, so it will be tested first
